etna/models/base.py

```diff
@@ -450,4 +467,303 @@ def get_model(self) -> Any:
         pass
 
 
-BaseModel = Union[PerSegmentModel, PerSegmentPredictionIntervalModel, MultiSegmentModel]
+class DeepBaseAbstractModel(ABC):
+    """Interface for etna native deep models."""
+
+    @abstractmethod
+    def forecast(self, ts: TSDataset, horizon: int, *args, **kwargs) -> TSDataset:
+        """Make predictions.
+
+        Parameters
+        ----------
+        ts:
+            Dataset with features and expected decoder length for context
+
+        horizon:
+            Horizon to predict for
+
+        Returns
+        -------
+        :
+            Dataset with predictions
+        """
+        pass
+
+    @abstractmethod
+    def make_samples(self, x: pd.DataFrame) -> Union[Iterator[dict], Iterable[dict]]:
+        """Make samples from input slices of TSDataset.
+
+        Parameters
+        ----------
+        x:
+            Slices are per-segment Dataframes
+
+        Returns
+        -------
+        :
+            Samples of input slices
+        """
+        pass
+
+    @abstractmethod
+    def raw_fit(self, ts: "Dataset") -> "DeepBaseAbstractModel":
+        """Fit model with torch like Dataset.
+
+        Parameters
+        ----------
+        ts:
+            Samples with data to fit on.
+
+        Returns
+        -------
+        :
+            Trained Model
+        """
+        pass
+
+    @abstractmethod
+    def raw_predict(self, ts: "Dataset") -> Dict[Tuple[str, str], np.ndarray]:
+        """Make inference on torch like Dataset.
+
+        Parameters
+        ----------
+        ts:
+            Samples with data to make inference on.
+
+        Returns
+        -------
+        :
+            Predictions for each segment
+        """
+        pass
+
+    @abstractmethod
+    def step(self, batch: dict, *args, **kwargs) -> Tuple["torch.Tensor", "torch.Tensor", "torch.Tensor"]:
+        """Make batch step.
+
+        Parameters
+        ----------
+        batch:
+            Batch with data to make inference on.
+
+        Returns
+        -------
+        :
+            loss, true_target, prediction_target
+        """
+        pass
+
+
+class DeepBaseModel(LightningModule, FitAbstractModel, DeepBaseAbstractModel, BaseMixin):
+    """Class for partially implemented interfaces."""
+
+    def __init__(
+        self,
+        *,
+        encoder_length: int,
+        decoder_length: int,
+        train_batch_size: int,
+        test_batch_size: int,
+        train_dataloader_kwargs: dict,
+        test_dataloader_kwargs: dict,
+        val_dataloader_kwargs: dict,
```

Comment:
We imply that val_batch_size is the same as test_batch_size?

Reply:
It seems it's all about perfomance and memory limits. 
We should get the same results regardless of batch size.